numpy/core/fromnumeric.py

```diff
@@ -604,15 +604,20 @@ def _transpose_dispatcher(a, axes=None):
 @array_function_dispatch(_transpose_dispatcher)
 def transpose(a, axes=None):
     """
-    Permute the dimensions of an array.
+    Reverse or permute the axes of an array; returns the modified array.
+    For an array a with two axes, transpose(a) gives the matrix transpose
+    of a.
 
     Parameters
     ----------
     a : array_like
         Input array.
-    axes : list of ints, optional
-        By default, reverse the dimensions, otherwise permute the axes
-        according to the values given.
+    axes : tuple or list of ints, optional
+        If not specified, this function reverses the order of the axes.
+        If specified, it must be a tuple, list or similar object which contains
```

Comment:
"similar object", is typically called "sequence" in python language. In fact, `tuple or list` can probably also be replaced with `sequence`. Unless we want to nudge people towards `tuple` specifically (since in many cases `axes` has to be a tuple).

Looks good to me.

Reply:
I would lean towards documenting it as just `tuple or list`, and not encouraging users to to pass anything else.